### ISSUE TITLE
Corrupted Keepnote Notebook

### ISSUE THREAD

**Issue by kinjikij** [mdrasmus/keepnote#744]
Lost all my notes today and I already understood how to recover them (clear these files and leave an empty <node></node> (node.xml) or <notebook></notebook> (notebook.nbk) xml construct in the files). But the issue is I dont see those folders in KeepNote directory.
I'll leave a picture of the situatsion and the log file.
Please help me get my notes back...

https://i.imgur.com/Jp2qS3i.png
![2019-12-26 16_57_08-KeepNote](https://user-images.githubusercontent.com/59257562/71480293-fabca700-2800-11ea-905b-f4cc8346cd2d.png)

**Comment by ransacrob** [comment]
Have you had any luck fixing this as yet?

**Comment by kinjikij** [comment]
Nope, changed to cherrytree

Hankige Outlook Androidi jaoks<https://aka.ms/ghei36>

________________________________

**Comment by ransacrob** [comment]
Thank you for your response. I don't suppose you were able to obtain your previous notes anyhow?

**Comment by kinjikij** [comment]
Nope, nothing worked for me and no support what so ever from them. If you find a solution please write it to me also 😊

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by ransacrob** [comment]
Will do - i, like you looked for all the solutions, but was unable to retrieve my notes, guess I will have to move on. Shame really.

**Comment by koshka-diana** [comment]
I have the exact same problem. I wish there was a way to fix this, I`ve put so much time into those notes.

**Comment by Badgerthelearner** [comment]
Hi all,
I had the same problem as you and I found and answer for my problem. I am sharing it in case you have the same problem.
1) I looked where my notebook was stored and I backed up all the files in a separate folder (you never know when you will be causing more problems!)
2) You will have different note.xml files in your notebook. Try starting with the one the main folder and if it doesn't work, keep moving through the folders until you find the corrupted node.
3) Open the node.xml in noted (textpad, notepad++ or whatever text editor you prefer)
4) There is a header and a footer called "node"
Eg
<?xml version="1.0" encoding="UTF-8"?>
<node>
<version>6</version>
<dict>
  <key>attr_tables</key><array>
    <dict>
      <key>name</key><string>Default Table</string>
      <key>key</key><string>default</string>
      <key>attrs</key><array>
        <string>title</string>
        <string>created_time</string>
        <string>modified_time</string>
.
.
.
</dict>
</node>
5) Remove everything between them and leave only
<?xml version="1.0" encoding="UTF-8"?>
<node>
</node>
6) Open your notebook again. Keepnote will regenerate the correct content in between. If this was the corrupted node then it will open as it used to.

I hope this helps!

**Comment by joyboyalc** [comment]
Hi, not sure if you're still looking for an answer or someone else might be but here's how I was able to find my notes. (Keep in mind this does not fix the corrupted keepnote rather it allows your to access your notes and be able to backup/migrate them.

I ran into the same problem you did, files corrupted cant open, also couldn't find my 'note' files in the KeepNote directory.

This was my first error:
<img width="256" alt="ErrorKeep" src="https://user-images.githubusercontent.com/111026687/184044674-6670e0c2-050d-42af-88e3-5cf4706e62e2.png">

I checked C:/Program Files (x86) and I didnt see my note files anywhere:
<img width="472" alt="FILESC" src="https://user-images.githubusercontent.com/111026687/184044756-6708c53c-677d-4bc7-9b22-b00beca0c7e3.png">

Turns out KeepNote saves the files on a different directory, to get to it you can click the windows key and type out %APPDATA% and click enter, from there you're going to want to click on 'Local' files, from there you navigate to 'VirtualStore' and in there click on Program Files (x86) then click on KeepNote. All your files should be here. Here is the entire path C:\Users\Your_User\AppData\Local\VirtualStore\Program Files (x86)\KeepNote
<img width="513" alt="APPDATA" src="https://user-images.githubusercontent.com/111026687/184045314-0ce12429-584c-4fd7-b696-55f3278a31f9.png">


<img width="470" alt="VirtualStore" src="https://user-images.githubusercontent.com/111026687/184045325-160c7055-4c94-4adb-9652-2caf2302fa81.png">

<img width="475" alt="ACD" src="https://user-images.githubusercontent.com/111026687/184045324-9d2f5cf0-fc52-4c14-90fb-cb6722ee3178.png">
